temporalio/worker/_worker.py

```diff
@@ -382,6 +409,13 @@ def __init__(
             use_worker_versioning=use_worker_versioning,
             disable_safe_workflow_eviction=disable_safe_workflow_eviction,
         )
+
+        root_plugin: Plugin = _RootPlugin()
+        for plugin in reversed(list(plugins)):
+            root_plugin = plugin.init_worker_plugin(root_plugin)
```

Comment:
Have to move all of this to the very top of the function, and move all of the validation of parameters and stuff below this and make sure we never reference a parameter directly after we run these Plugins can change any of these parameters.